sql: fix LSC that incorrectly missed secondary index rewrite during ALTER PK

Xiang-Gu opened the pull request:
Fixes #114436

Release note (bug fix): Previously, when set `use_declarative_schema_changer=off`
and we attempt to alter primary key on a table that has unique secondary
indexes on new-pk-cols, the unique secondary index would incorrectly
still have old-pk-cols as its keySuffixColumn after the ALTER PK. This
is problematic because a subsequent dropping of the old-pk-cols would
unexpectedly drop those unique secondary indexes as well, even without
CASCADE. This commit fixes that.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/114622)
<!-- Reviewable:end -->

Xiang-Gu committed:
sql: fix LSC that incorrectly missed secondary index rewrite during ALTER PK

Fixes #114436

Release note (bug fix): Previously, when set `use_declarative_schema_changer=off`
and we attempt to alter primary key on a table that has unique secondary
indexes on new-pk-cols, the unique secondary index would incorrectly
still have old-pk-cols as its keySuffixColumn after the ALTER PK. This
is problematic because a subsequent dropping of the old-pk-cols would
unexpectedly drop those unique secondary indexes as well, even without
CASCADE. This commit fixes that.

fqazi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> Nice find!

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/114622#-:-NjmXlyP4q9YgRBO5Q4I:bcdfsno)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @Xiang-Gu)



<!-- Sent from Reviewable.io -->

Xiang-Gu commented:
TFTR!

bors r+
